server/lib/konnector_poller.coffee

```diff
@@ -18,187 +18,154 @@ month = 30 * day
                         savedKonnector.removeEncryptedFields fields
-
                         # Create/Update lastAutoImport in database
-                        savedKonnector.updateAttributes data, (err) =>
+                        savedKonnector.updateAttributes data, (err, body) ->
                             if err
                                 log.error err
+                        @create konnector, @findNextUpdate(konnector)
+                        cb() if cb?
 
-                    @create konnector, diff
-
-    create: (konnector, diff) ->
-
-        # if diff is present and valid
-        if diff > 0
-            interval = diff
-        else
-            interval = periods[konnector.importInterval]
-
-        # Check if interval value is more than 10 sec
-        # And if the value is in the periods list
-        if interval? and interval > 10000
-            @prepareNextCheck konnector, interval
-        else
-            log.info "konnector #{konnector.slug} has an " +
-            "incorrect importInterval value"
-
-    prepareNextCheck: (konnector, interval) ->
-
-        ## dirty hack for bypassing timeout max value
-
-        # If time exists, assign that value to interval
-        if konnector.time?
-            interval = konnector.time
-
-        # if interval is more than 23 days,
-        # save the excess in konnector['time']
-        if interval > (23 * day)
-            konnector['time'] = interval - (23 * day)
-            interval = 23 * day
-        else
-            # if interval value is less than 23 days,
-            if konnector.time?
-                # time is no longer needed
-                delete konnector.time
+    create: (konnector, nextUpdate) ->
 
-        @createTimeout konnector, interval
-
-    createTimeout: (konnector, interval) ->
         now = moment()
-        nextUpdate = now.clone()
-        nextUpdate = now.add interval, 'ms'
+        nextUpdates[konnector.slug] = [nextUpdate, konnector]
+        interval = nextUpdate.diff now.clone(), 'ms'
+        if interval < day
+            @createTimeout konnector, interval
+
+    prepareNextCheck: () ->
+        for slug in Object.keys(nextUpdates)
+            # If interval is more than 1 day,
+            # timeout isn't started (prepareNextCheck call every day)
+            [nextUpdate, konnector]  = nextUpdates[slug]
+            now = moment()
+            interval = nextUpdate.diff now.clone(), 'ms'
+            if interval < day
+                @createTimeout konnector, interval
 
-        log.info "Next check of konnector #{konnector.slug} on " +
-        "#{nextUpdate.format(format)}"
-        # Create the timeout and place it timeouts
+    createTimeout: (konnector, interval) ->
         timeouts[konnector.slug] = setTimeout @checkImport.bind(@, konnector, interval), interval
 
-    checkImport: (reference, interval) ->
-        # retrieves the last version of the konnector
-        Konnector.find reference.id, (err, konnector) =>
```

Comment:
You removed that, are you sure when the code executes the import, it has the latest version of the konnectors? Anyway, it'd be better to do it in the "importer" rather than here, so it's good to remove it from here.